File: HumanPlayer.java
import java.util.*;

/**
 * Translated from the original python written by Christine Alvorado
 */

public class HumanPlayer extends Player {

    public static Scanner sc = new Scanner(System.in);

    // Constructor    
    public HumanPlayer(int playerNum, int oppNum) {
        super(playerNum, oppNum, 0);
    }

    public int chooseMove(BoardGame board) {
        // Returns the next move that this player wants to make
        int move = -1;
        System.out.println("Please enter your move:");
        move = sc.nextInt();
        while (!board.legalMove(this, move)) {
            System.out.println("" + move + "is not valid");
            System.out.println("Please enter your move:");
            move = sc.nextInt();
        }
        return move;
    }
} 

File: csharp/Axiom/Players/HumanPlayer.cs
using System;
namespace Axiom
{
    public class HumanPlayer : Player
    {

        // Constructor
        public HumanPlayer(Color player, Color opp) : base(player, opp) { }

        override
        public int ChooseMove(BoardGame board)
        {
            // Returns the next move that this player wants to make
            int move = -1;
            Console.WriteLine("Please enter your move:");
            move = Convert.ToInt32(Console.ReadLine());
            while (!board.LegalMove(this, move))
            {
                Console.WriteLine("" + move + "is not valid");
                Console.WriteLine("Please enter your move:");
                move = Convert.ToInt32(Console.ReadLine());
            }
            return move;
        }
    }
}

